framework/src/partitioner/LibmeshPartitioner.C

```diff
@@ -123,14 +135,51 @@ return libmesh_make_unique<MetisPartitioner>();
 }
 
 void
+LibmeshPartitioner::prepare_blocks_for_subdomain_partitioner(SubdomainPartitioner & subdomain_partitioner)
+{
+  auto group_begin = _subdomain_blocks.begin();
+  auto group_end = _subdomain_blocks.end();
+
+  subdomain_partitioner.chunks.clear();
+  for (auto group = group_begin; group != group_end; ++group)
+  {
+     std::set<subdomain_id_type> subdomain_ids;
+     auto subdomain_ids_vec = _mesh.getSubdomainIDs(*group);
+     auto subdomain_begin = subdomain_ids_vec.begin();
+     auto subdomain_end = subdomain_ids_vec.end();
+     for (auto subdomain_id = subdomain_begin; subdomain_id != subdomain_end; ++subdomain_id)
+     {
+       subdomain_ids.insert(*subdomain_id);
+     }
+     subdomain_partitioner.chunks.push_back(subdomain_ids);
+  }
+}
+
+void
 LibmeshPartitioner::partition(MeshBase &mesh, const unsigned int n)
 {
+  if (_partitioner_name == "subdomain_partitioner")
+  {
+    if (_partitioner.get())
+      prepare_blocks_for_subdomain_partitioner(dynamic_cast<SubdomainPartitioner&>(*_partitioner.get()));
```

Comment:
This is still not logically correct. The reason is that dynamic casts can fail, but you are casting to a reference type. That just doesn't make sense. If you know the cast will succeed use a static cast instead.